storage/src/main/java/org/apache/kafka/storage/internals/log/LogLoader.java

```diff
@@ -169,6 +169,7 @@ public LoadedLogOffsets load() throws IOException {
                     long offset = LogFileUtils.offsetFromFile(file);
                     if (offset >= minSwapFileOffset && offset < maxSwapFileOffset) {
                         logger.info("Deleting segment files {} that is compacted but has not been deleted yet.", file.getName());
+                        @SuppressWarnings("UnusedLocalVariable")
                         boolean ignore = file.delete();
```

Comment:
Shouldn't we actually check if the file was deleted?

@mimaison I see you worked the Java conversion of this class. WDYT about checking the result of `delete` and `renameTo` below?